sys-design/interfaces.go

```diff
@@ -0,0 +1,230 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"strconv"
+)
+
+// Device interface
+type Device interface {
+	Name() string
+	TogglePower()
+	NumberPad(value int)
+	IsOn() bool
+}
+
+// TV struct
+type TV struct {
+	isOn    bool
+	channel int
+}
+
+func (tv *TV) TogglePower() {
+	if tv.isOn {
+		fmt.Println("TV Power off")
+		tv.isOn = false
+	} else {
+		fmt.Println("TV Power on")
+		tv.isOn = true
+	}
```

Comment:
Optional enhancement - This can be converted into another function and can be reused across devices.
Similarly for NumberPad